<?xml version="1.0"?>

<!-- dalsa_genie_nano_c2420 launch file -->
<launch>

<arg name="dalsa_camera_topic"      default="dalsa_camera"/>
<arg name="publish_mono"            default="false"/>
<arg name="dalsa_camera_mono_topic" default="dalsa_camera_mono"/>
<arg name="ndvi"                    default="false"/>
<arg name="throttle"                default="true"/>


  <!--launch the driver -->
  <node pkg="dalsa_genie_nano_c2420" type="dalsa_genie_nano_c2420_driver" name="dalsa_genie_nano_c2420_driver" output="screen" respawn="true">
    <param name="dalsa_camera_frame"               value="dalsa_link"/>
    <param name="dalsa_camera_topic"               value="$(arg dalsa_camera_topic)"/>
    <param name="camera_index"                     value="0"/>
    <param name="publish_mono"                     value="$(arg publish_mono)"/>
    <param name="dalsa_camera_mono_topic"          value="$(arg dalsa_camera_mono_topic)"/>
    <param name="use_synchronous_buffer_cycling"   value="false"/> <!--make it "true" at your own risk! -->
    <param name="tune_streaming_threads"           value="false"/> <!--make it "true" at your own risk!-->
    <param name="turbo_mode"                       value="false"/> <!--make it "true" at your own risk!-->
  </node>

 <!--publish compressed colored image (full and 720p-->
 <node pkg="image_transport" type="republish" name="dalsa_image_compress" args="raw in:=$(arg dalsa_camera_topic) compressed out:=$(arg dalsa_camera_topic)"/>
 <node pkg="image_transport" type="republish" name="dalsa_image720p_compress" args="raw in:=$(arg dalsa_camera_topic)_720p compressed out:=$(arg dalsa_camera_topic)_720p"/>

<!--publish compressed mono image (if used)-->
<group if="$(arg publish_mono)">
	<node pkg="image_transport" type="republish" name="dalsa_mono_image_compress" args="raw in:=$(arg dalsa_camera_mono_topic) compressed out:=$(arg dalsa_camera_mono_topic)"/>
  <node pkg="image_transport" type="republish" name="dalsa_mono_image720p_compress" args="raw in:=$(arg dalsa_camera_mono_topic)_720p compressed out:=$(arg dalsa_camera_mono_topic)_720p"/>
</group> 

<!-- Launch camera info  -->
<node pkg="dalsa_genie_nano_c2420" type="camera_info_publisher.py"  name="dalsa_camera_info">
  <param name="camera_info_path" value="$(find dalsa_genie_nano_c2420)/config/camera_info.yaml"/>
</node>

<!-- Start ndvi node (if used) -->
<group if ="$(arg ndvi)">
    <include file = "$(find dalsa_genie_nano_c2420)/launch/ndvi.launch" />
</group>

<!-- Throttle camera to 0.5Hz (if used) -->
<group if = "$(arg throttle)">
      <node name="dalsa_throttle" type="throttle" pkg="topic_tools" args="messages /dalsa_camera_720p/compressed 0.5 /dalsa_camera_720p/throttle/compressed" />
</group>

</launch>
